API/controlsClass/controlsClass.m

```diff
@@ -1,249 +1,243 @@
+        targetValue = 1
+        numGenerations = 500
 
         %(3) Nested Sampler (for Matlab version. There will be more options
-        %with the C version)
+        % with the C version)
         Nlive = 150
         Nmcmc = 0
-        propScale = 0.1     %Used if MCMC is used
-        nsTolerance = 0.1     %Target stopping tolerance
+        propScale = 0.1     % Used if MCMC is used
+        nsTolerance = 0.1     % Target stopping tolerance
 
         %(4) DREAM
         nSamples = 50000;          % Total number of samples
-        nChains = 10;               % Number of MCMC chains..
-        lambda = 0.5;               % Jump probabilities
-        p_unit_gamma = 0.2;
-        boundHandling = 'fold';     % Boundary handling
-        % prior = true;
-
+        nChains = 10               % Number of MCMC chains..
+        lambda = 0.5               % Jump probabilities
+        pUnitGamma = 0.2
+        boundHandling = 'fold'     % Boundary handling
     end
     
     %------------------------- Set and Get ------------------------------
     methods
         function obj = set.parallel(obj,val)
             if ~strcmpi(val,{'single', 'points', 'contrasts', 'all'})
-                error('Type must be ''single'', ''points'', ''contrasts'', ''all'' ');
+                throw(invalidOption('Type must be ''single'', ''points'', ''contrasts'', ''all'''));
             end
-
             obj.parallel = val;
         end
         
         function obj = set.procedure(obj,val)
             if ~strcmpi(val,{'calculate','simplex','DE', 'NS','dream'})
-                error('Type must be ''calculate'', ''simplex'', ''DE'', ''dream'' or ''NS'' ');
+                throw(invalidOption('Type must be ''calculate'', ''simplex'', ''DE'', ''dream'' or ''NS'''));
             end
-            
             obj.procedure = val;
         end
         
-        
         function obj = set.calcSldDuringFit(obj,val)
-            if ~strcmpi(val,{'yes','no'})
-                error('Type must be ''yes'' or ''no'' ');
+            if ~islogical(val)
+                throw(invalidValue('Type must be logical ''true'' or ''false'''));
             end
-            
-%             if strcmpi(val,'no')
-%                 warning('Will automatically default to ''yes'' if resample is set in problemDef')
-%             end
             obj.calcSldDuringFit = val;
         end
+
+        function obj = set.display(obj,val)
+            if ~strcmpi(val,{'off','iter','notify','final'})
+                throw(invalidOption('Display must be set to ''off'', ''iter'', ''notify'' or ''final'' '));
+            end
+            obj.display = val;
+        end 
+
+        function obj = set.resamPars(obj,val)
+            if length(val) ~= 2
+                throw(invalidValue('resamPars must have length of 2'));
+            end
+            if (~isnumeric(val(1)) || val(1) < 0 || val(1) > 1)
+                throw(invalidValue('resamPars(0) must be between 0 and 1'));
```

Comment:
Hmm, this exception (and others like it) combines two together. The `isnumeric` part would raise an `invalidType` exception, but the remaining clauses raise the `invalidValue` exception. It's probably ok to keep it like this, but might be worth thinking about.